File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02702/p02702_num6_parsed.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    string S;
    cin >> S;

    int count = 0;
    int len = S.length();

    for (int i = 0; i < len; i++) {
        int num = 0;
        for (int j = i; j < len; j++) {
            num = num * 10 + (S[j] - '0');
            if (num % 2019 == 0) {
                count++;
            }
        }
    }

    cout << count << endl;
    
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02702/p02702_num6_parsed.py
import sys

def count_pairs(S):
    count = 0
    for i in range(len(S)):
        num = 0
        for j in range(i, len(S)):
            num = num * 10 + int(S[j])
            if num % 2019 == 0:
                count += 1
    return count

S = sys.stdin.readline().strip()
print(count_pairs(S))